NDTensors/src/arraytensor/array.jl

```diff
@@ -61,8 +61,6 @@ end
 function permutedims!(
   output_array::MatrixOrArrayStorage, array::MatrixOrArrayStorage, perm, f::Function
 )
-  output_array = permutedims!!(
-    leaf_parenttype(output_array), output_array, leaf_parenttype(array), array, perm, f
-  )
+  output_array = permutedims!!(output_array, array, perm, f)
```

Comment:
Should this use `expose`?

Reply:
No so `permutedims!!` as I have written will call `permutedims!(expose(output), expose(array), perm, f)` in case the type of `output` or `array` needs to be modified.

Reply:
Thanks, I think I got confused and was looking at an old version of the code, makes sense.